From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Sheetal <sheetal@nvidia.com>
Date: Tue, 13 May 2025 08:57:20 +0000
Subject: [PATCH 212/240] NVIDIA: SAUCE: ASoC: Fix pll clock issue

Issue:
 Using 'rt5640-codec-sysclk-bclk1' compatible throws below error
 while running any of the rt5640 usecase,
   rt5640 3-001c: Unsupported clock setting 48000 for DAI 0
   rt5640 3-001c: ASoC: error at snd_soc_dai_hw_params on rt5640-aif1: -22

Fix:
  aud_mclk was used uninitialized that was leading to above error.
  Initialized aud_mclk as srate * 256.

Fixes: 9225307cecaa("NVIDIA: SAUCE: ASoC: tegra: support few external codecs")

Bug 5197062

Change-Id: I3a0b5ba826107e11d3fd69d3e1fb636e4c60a11c
Signed-off-by: Sheetal <sheetal@nvidia.com>
Reviewed-on: https://git-master.nvidia.com/r/c/3rdparty/canonical/linux-noble/+/3363136
Reviewed-by: Mohan kumar <mkumard@nvidia.com>
Reviewed-by: svcacv <svcacv@nvidia.com>
Reviewed-by: Sameer Pujar <spujar@nvidia.com>
GVS: buildbot_gerritrpt <buildbot_gerritrpt@nvidia.com>
(cherry picked from commit 818ddb04bc839d09c56f93b774a457dd24533f35)

diff --git a/sound/soc/tegra/tegra_codecs.c b/sound/soc/tegra/tegra_codecs.c
index 1386412dbd6be..d9146f1d8be6a 100644
--- a/sound/soc/tegra/tegra_codecs.c
+++ b/sound/soc/tegra/tegra_codecs.c
@@ -1,5 +1,5 @@
 // SPDX-License-Identifier: GPL-2.0-only
-// SPDX-FileCopyrightText: Copyright (c) 2021-2024 NVIDIA CORPORATION & AFFILIATES. All rights reserved.
+// SPDX-FileCopyrightText: Copyright (c) 2021-2025 NVIDIA CORPORATION & AFFILIATES. All rights reserved.
 //
 // tegra_codecs.c - External audio codec setup
 
@@ -281,6 +281,9 @@ int tegra_codecs_runtime_setup(struct snd_soc_pcm_runtime *rtd,
 
 	vrtd = find_rtd(rtd, "rt565x-codec-sysclk-bclk1");
 	if (vrtd) {
+		props = simple_priv_to_props(simple, vrtd->id);
+		aud_mclk = props->mclk_fs * srate;
+
 		err = set_pll_sysclk(vrtd, RT5659_PLL1_S_BCLK1, RT5659_SCLK_S_PLL1,
 				     srate, channels, width, aud_mclk);
 		if (err < 0)
@@ -290,6 +293,9 @@ int tegra_codecs_runtime_setup(struct snd_soc_pcm_runtime *rtd,
 
 	vrtd = find_rtd(rtd, "rt5640-codec-sysclk-bclk1");
 	if (vrtd) {
+		props = simple_priv_to_props(simple, vrtd->id);
+		aud_mclk = props->mclk_fs * srate;
+
 		err = set_pll_sysclk(vrtd, RT5640_PLL1_S_BCLK1, RT5640_SCLK_S_PLL1,
 				     srate, channels, width, aud_mclk);
 		if (err < 0)
-- 
2.43.0

